crates/net/src/http/headers.rs

```diff
@@ -37,19 +29,33 @@ impl Headers {
 
     /// This method appends a new value onto an existing header, or adds the header if it does not
     /// already exist.
-    pub fn append(&self, name: &str, value: &str) {
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use gloo_net::http::Headers;
+    /// # fn no_run() {
+    /// let headers = Headers::new();
+    /// headers.append("Content-Type", "text/plain");
+    /// assert_eq!(headers.get("Content-Type"), Some("text/plain".to_string()));
+    ///
+    /// headers.append("Content-Type", "text/html");
+    /// assert_eq!(headers.get("Content-Type"), Some("text/plain, text/html".to_string()));
+    /// # }
+    /// ```
+    pub fn append(&mut self, name: &str, value: &str) {
```

Comment:
I know `web_sys::Headers` (and all? `web_sys` types)  allow inner mutability through a `&T` but I'm not sure if it's idiomatic to do the same on our wrapper types. 

For example, if someday we choose to change the implementation of `Header` wrap a `HashSet<&str, String>`  this method would need to be `&mut` or we're committing to using a `RefCell<_>` forever